testing/integration/testdata/.upgrade-test-agent-versions.yml

```diff
@@ -5,8 +5,8 @@
 # upgrade integration tests.
 
 testVersions:
-  - 8.16.0-SNAPSHOT
-  - 8.15.3
+  - 8.16.1
```

Comment:
This does not look right. It's missing 8.16.2-SNAPSHOT according to this https://snapshots.elastic.co/latest/8.16.json

I'll into this later.

Reply:
The 8.16.2 snapshot's manifest is quite fresh, perhaps we need to retry later https://snapshots.elastic.co/8.16.2-aeae87d8/manifest-8.16.2-SNAPSHOT.json

But I expected the previous snapshot to be available too.

Reply:
False alarm, this actually makes sense since all these versions are coming from the same `8.16` branch we tests against only the released versions.